pkg/operation/botanist/controlplane/konnectivity/konnectivity_server.go

```diff
@@ -0,0 +1,560 @@
+							},
+							LivenessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path:   "/healthz",
+										Port:   intstr.IntOrString{Type: intstr.Int, IntVal: healthPort},
+										Host:   "127.0.0.1",
+										Scheme: corev1.URISchemeHTTP,
+									},
+								},
+							},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("30m"),
+									corev1.ResourceMemory: resource.MustParse("40Mi"),
+								},
+								Limits: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("300m"),
+									corev1.ResourceMemory: resource.MustParse("400Mi"),
+								},
+							},
+							Ports: []corev1.ContainerPort{{
+								Name:          "server",
+								ContainerPort: ServerHTTPSPort,
+								HostPort:      ServerHTTPSPort,
+							}, {
+								Name:          "agent",
+								ContainerPort: ServerAgentPort,
+								HostPort:      ServerAgentPort,
+							}, {
+								Name:          "admin",
+								ContainerPort: adminPort,
+								HostPort:      adminPort,
+							}, {
+								Name:          "health",
+								ContainerPort: healthPort,
+								HostPort:      healthPort,
+							}},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      k.secrets.Server.Name,
+									MountPath: serverMountPath,
+								},
+								{
+									Name:      k.secrets.ClientCA.Name,
+									MountPath: clientCAMountPath,
+								},
+								{
+									Name:      k.secrets.Kubeconfig.Name,
+									MountPath: kubeconfigMountPath,
+								},
+							},
+						}},
+						Volumes: []corev1.Volume{
+							{
+								Name: k.secrets.Server.Name,
+								VolumeSource: corev1.VolumeSource{
+									Secret: &corev1.SecretVolumeSource{
+										SecretName: k.secrets.Server.Name,
+									},
+								},
+							},
+							{
+								Name: k.secrets.ClientCA.Name,
+								VolumeSource: corev1.VolumeSource{
+									Secret: &corev1.SecretVolumeSource{
+										SecretName: k.secrets.ClientCA.Name,
+									},
+								},
+							},
+							{
+								Name: k.secrets.Kubeconfig.Name,
+								VolumeSource: corev1.VolumeSource{
+									Secret: &corev1.SecretVolumeSource{
+										SecretName: k.secrets.Kubeconfig.Name,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		gateway = &networkingv1beta1.Gateway{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: k.Namespace,
+				Labels:    getLabels(),
+			},
+			Spec: istionetworkingv1beta1.Gateway{
+				Selector: k.IstioIngressLabels,
+				Servers: []*istionetworkingv1beta1.Server{{
+					Hosts: k.Hosts,
+					Port: &istionetworkingv1beta1.Port{
+						Number:   serverAgentPortUnsigned,
+						Protocol: "TLS",
+						Name:     "tls-tunnel",
+					},
+					Tls: &istionetworkingv1beta1.ServerTLSSettings{
+						Mode: istionetworkingv1beta1.ServerTLSSettings_PASSTHROUGH,
```

Comment:
The istio [Gateway go types](https://github.com/istio/api/blob/master/networking/v1beta1/gateway.pb.go) have the TLS mode as an int32 - you use the constant here.
However, the CRD for the Gateway we use [is a string](https://github.com/gardener/gardener/blob/e37bb3d5c8b4209702d90d71e59d172c55129d57/charts/istio/istio-crds/templates/crd-all.gen.yaml#L1546). 
I initially thought that this will be converted internally, however checking the Gateway resource in the cluster shows that the TLS mode PASSTHROUGH is missing on the Konnectivity server gateway.

>   
      name: tls-tunnel
      number: 8132
      protocol: TLS
    tls: {}
    
Vs. for the kube-apiserver
    
>
      name: tls
      number: 443
      protocol: TLS
    tls:
      mode: PASSTHROUGH


I cannot definitely say that this causes  TLS to be terminated as the istio-ingressgateway or if Passthrough is the default setting when nothing is specified.
However,  a tunnel connection cannot be established so this is a possible reason.

Reply:
This doesn't matter - the default value for PASSTHROUGH is 0. The TLS object is empty (but not nil) and therefore this is a PASSTHROUGH cluster -  https://github.com/istio/istio/blob/65bd2ac632cbd6ec8e3fd85bb9874f685aea0cfd/pkg/config/gateway/gateway.go#L45-L57